acme/client.py

```diff
@@ -0,0 +1,234 @@
+        response = requests.request(method, url, data=data, headers=all_headers)
+
+        if response.status_code == 200:
+
+            # Return JSON if that's what it is (this should be the default)
+            if response.headers.get('Content-Type', str) == 'application/json':
+                return response.json()
+            
+            return response
+
+        else:
+            raise Exception(f"Request failed with status code {response.status_code}: {response.text}")
+
+
+    def login(self, email: str, password: str):
+        """Log the user into ME.org.
+
+        Parameters
+        ----------
+        email : str
+            Registered email address.
+        password : str
+            Password (will be hashed)
+
+        Raises
+        ------
+        Exception
+            When the login fails. 
+        """
+
+        # Assemble payload
+        login_data = {
+            'email': email,
+            'password': hl.sha256(password.encode('UTF-8')).hexdigest(),
+            'hashed': 'true'
+        }
+
+        # Call
+        response = self._make_request('post', endpoint='login', data=login_data)
+
+        # Successful login
+        if response.get('status', 'error') == 'success':
+            auth_headers = {
+                'X-User-Id': response['data']['userId'],
+                'X-Auth-Token': response['data']['authToken']
+            }
+
+            self.headers.update(auth_headers)
+
+        # Unsuccessful login
+        else:
+
+            raise Exception("Login failed")
+
+
+    def get_file_status(self, file_id: str) -> dict:
+        """Get the file status.
+
+        Parameters
+        ----------
+        file_id : str
+            ID of the file.
+
+        Returns
+        -------
+        dict or requests.Response
+            Response from ME.org.
+        """
+        return self._make_request(
+            method='get',
+            endpoint='/files/status/{file_id}',
+            data=dict(file_id=file_id)
+        )
+
+
+    def upload_file(self, file_path):
+        raise NotImplementedError()
+
+
+    def logout(self):
+        """Log the user out. Likely not necessary, can just let sessions expire.
+
+        Returns
+        -------
+        dict or requests.Response
+            Response from ME.org.
+        """
+        return self._make_request('post', endpoint='logout')
+    
+
+    def get_model_outputs(self, model_output_id):
+        """Get a list of model outputs.
+
+        Parameters
+        ----------
+        model_output_id : str
+            Model output
+
+        Returns
+        -------
+        dict or requests.Response
+            Response from ME.org.
+        """
+        return self._make_request(
+            method='get',
+            endpoint='modeloutput/{model_output_id}/files',
+            data=dict(model_output_id=model_output_id)
+        )
+
+
+    def start_analysis(self, model_output_id: str) -> dict:
```

Comment:
We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def start_analysis(self, model_output_id: str) -> Union[dict, requests.Response]:
```